external/storm-kafka/src/jvm/org/apache/storm/kafka/KafkaUtils.java

```diff
@@ -213,7 +213,9 @@ public static ByteBufferMessageSet fetchMessages(KafkaConfig config, SimpleConsu
                 throw new TopicOffsetOutOfRangeException(msg);
             } else {
                 String message = "Error fetching data from [" + partition + "] for topic [" + topic + "]: [" + error + "]";
-                LOG.error(message);
+                if (!error.equals(KafkaError.NOT_LEADER_FOR_PARTITION)) {
```

Comment:
I would still suggest to remove this log completely since we are refreshing for all such errors and a warning with the exception should suffice.